tests/by-util/test_dd.rs

```diff
@@ -1528,3 +1528,62 @@ fn test_multiple_processes_reading_stdin() {
         .succeeds()
         .stdout_only("def\n");
 }
+
+#[test]
+#[cfg(all(unix, not(target_os = "macos"), not(target_os = "freebsd")))]
+fn test_reading_partial_blocks_from_fifo() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    at.mkfifo("fifo");
+    let fifoname = at.plus_as_string("fifo");
+
+    let child = ucmd
+        .args(&["ibs=3", "obs=3", &format!("if={}", fifoname)])
+        .set_stderr(Stdio::piped())
+        .run_no_wait();
+
+    UCommand::new()
+        .arg(format!(
+            "({TESTS_BINARY} printf 'ab'; {TESTS_BINARY} sleep 0.1; {TESTS_BINARY} printf 'cd') > {fifoname}"
```

Comment:
These tests are failing for reasons I don't understand yet. It seems that the two bytes "ab" and the two bytes "cd" are sometimes being read by `dd` one byte at a time, resulting in one partial block read comprising the byte "a" and a separate partial block read comprising the byte "b" (or "c" and "d", respectively). It seems to be happening only when I run a large set of tests, like `cargo test test_dd`, and not when I run the tests individually or as a pair, as in `cargo test test_dd::test_reading_partial_blocks_from_fifo`.

Reply:
I switched to using explicit `Command`, let's see if that helps.